src/freenet/clients/fcp/ClientGet.java

```diff
@@ -344,7 +355,7 @@ public void onLostConnection(ClientContext context) {
 	@Override
 	public void onSuccess(FetchResult result, ClientGetter state) {
 		Logger.minor(this, "Succeeded: "+identifier);
-		Bucket data = result.asBucket();
+		Bucket data = binaryBlob ? state.getBlobBucket() : result.asBucket();
```

Comment:
Not just that. It is also needed for ReturnType=direct. We want to return the binary blob, not the fetched data. The FetchResult contains the fetched data, which here will be a NullBucket, not the binary blob.